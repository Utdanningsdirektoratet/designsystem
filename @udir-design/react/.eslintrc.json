{
  "extends": ["plugin:@nx/react", "../../.eslintrc.json"],
  "ignorePatterns": ["!**/*", "storybook-static"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "parserOptions": {
        "project": [
          "@udir-design/react/tsconfig.*?.json",
          "@udir-design/react/tsconfig.storybook.json"
        ]
      },
      "rules": {}
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    }
  ]
}
