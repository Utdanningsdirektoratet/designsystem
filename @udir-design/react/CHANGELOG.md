## 0.1.0-1 (2024-10-22)

build: don't bundle depenencies


## 0.1.0-0 (2024-10-22)

### ğŸš€ Features

- add Button component ([8d698eb](https://github.com/Utdanningsdirektoratet/designsystem/commit/8d698eb))
