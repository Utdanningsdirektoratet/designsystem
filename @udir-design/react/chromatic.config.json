{
  "onlyChanged": true,
  "projectId": "Project:667e8f07bf467ff4403dfe77",
  "storybookBaseDir": "@udir-design/react",
  "zip": true
}
