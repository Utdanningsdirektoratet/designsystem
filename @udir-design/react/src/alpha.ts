/*
 * This is the entry point for the /alpha sub-module.
 * It exports the alpha-stage components, plus everything in beta (& stable).
 */
export * from './beta';
export * from './components/alpha';
export * from './utilities/hooks/alpha';
