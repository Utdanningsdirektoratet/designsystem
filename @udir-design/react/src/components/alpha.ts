/*
 * This file exports the components that are in the alpha stage of development
 */

export * from './button/Button';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './details/Details';
export * from './dialog/Dialog';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './errorSummary/ErrorSummary';
export * from './field/Field';
export * from './fieldset/Fieldset';
export * from './input/Input';
export * from './link/Link';
export * from './list/List';
export * from './loaders';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skipLink/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
export * from './textarea/Textarea';
export * from './textfield/Textfield';
export * from './toggleGroup/ToggleGroup';
export * from './tooltip/Tooltip';
export * from './typography';
