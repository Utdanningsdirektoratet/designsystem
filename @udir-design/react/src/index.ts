export * from './button/Button';
export * from './accordion/Accordion';
export * from './alert/Alert';
export * from './avatar/Avatar';
export * from './badge/Badge';
export * from './breadcrumbs/Breadcrumbs';
export * from './card/Card';
export * from './checkbox/Checkbox';
export * from './chip/Chip';
export * from './divider/Divider';
export * from './dropdown/Dropdown';
export * from './error-summary/ErrorSummary';
export * from './fieldset/Fieldset';
export * from './help-text/HelpText';
export * from './link/Link';
export * from './list/List';
export * from './loaders/index';
export * from './modal/Modal';
export * from './pagination/Pagination';
export * from './popover/Popover';
export * from './radio/Radio';
export * from './search/Search';
export * from './select/Select';
export * from './skip-link/SkipLink';
export * from './switch/Switch';
export * from './table/Table';
export * from './tabs/Tabs';
export * from './tag/Tag';
