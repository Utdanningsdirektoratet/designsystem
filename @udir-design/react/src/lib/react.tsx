import styles from './react.module.scss';

export const React: React.FC = () => (
  <div className={styles['container']}>
    <h1>Welcome to React!</h1>
  </div>
);

export default React;
