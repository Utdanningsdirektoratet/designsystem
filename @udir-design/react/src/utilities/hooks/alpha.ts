/*
 * This file exports the hooks that are in the alpha stage of development
 */

export {
  type UsePaginationProps,
  usePagination,
} from './usePagination/usePagination';
export {
  useTableOfContents,
  type useTableOfContentsProps,
} from './useTableOfContents/useTableOfContents';
