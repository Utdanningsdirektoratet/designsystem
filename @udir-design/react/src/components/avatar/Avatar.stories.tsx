import type { Meta, StoryObj } from '@storybook/react-vite';
import { Avatar, Badge, Dropdown, Link } from '@udir-design/react/alpha';
import { BriefcaseIcon, ChevronUpIcon } from '@navikt/aksel-icons';

const meta: Meta<typeof Avatar> = {
  component: Avatar,
  tags: ['alpha'],
  parameters: {
    layout: 'padded',
    customStyles: {
      display: 'flex',
      gap: 'var(--ds-size-2)',
      justifyContent: 'center',
      alignItems: 'center',
    },
  },
};

const profileImage =
  'https://images.unsplash.com/photo-1633332755192-727a05c4013d?q=80&w=1480&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D';

export default meta;
type Story = StoryObj<typeof Avatar>;

export const Preview: Story = {
  args: {
    'aria-label': 'Ola Nordmann',
    children: '',
  },
};

export const Content: Story = {
  args: { 'aria-label': 'Ola Nordmann' },
  render: (args) => (
    <>
      <Avatar {...args} />
      <Avatar {...args} initials="ON" />
      <Avatar {...args}>
        <BriefcaseIcon />
      </Avatar>
      <Avatar {...args}>
        <img src={profileImage} alt="" />
      </Avatar>
    </>
  ),
};

export const Sizes: Story = {
  render: (args) => (
    <>
      <Avatar {...args} data-size="xs" aria-label="extra small" initials="xs" />
      <Avatar {...args} data-size="xs" aria-label="extra small" />
      <Avatar {...args} data-size="sm" aria-label="small" initials="sm" />
      <Avatar {...args} data-size="sm" aria-label="small" />
      <Avatar {...args} data-size="md" aria-label="medium" initials="md" />
      <Avatar {...args} data-size="md" aria-label="medium" />
      <Avatar {...args} data-size="lg" aria-label="large" initials="lg" />
      <Avatar {...args} data-size="lg" aria-label="large" />
    </>
  ),
};

export const ColorVariants: Story = {
  render: (args) => (
    <>
      <Avatar {...args} data-color="neutral" aria-label="color neutral" />
      <Avatar {...args} data-color="accent" aria-label="color accent" />
      <Avatar {...args} data-color="support1" aria-label="color support1" />
      <Avatar {...args} data-color="support2" aria-label="color support2" />
    </>
  ),
};

export const InDropdown: Story = {
  parameters: {
    layout: 'fullscreen',
    customStyles: {
      height: '320px',
    },
  },
  render: (args) => (
    <Dropdown.TriggerContext>
      <Dropdown.Trigger variant="tertiary">
        <Avatar
          {...args}
          aria-label="Ola Nordmann"
          data-size="sm"
          initials="ON"
        />
        Ola Nordmann
        <ChevronUpIcon aria-hidden />
      </Dropdown.Trigger>
      <Dropdown placement="bottom-end" data-size="md" open>
        <Dropdown.Heading>Velg profil</Dropdown.Heading>
        <Dropdown.List>
          <Dropdown.Item>
            <Dropdown.Button>
              <Badge.Position overlap="circle">
                <Badge data-color="danger" data-size="sm" />
                <Avatar
                  {...args}
                  aria-label="Ola Nordmann"
                  data-size="xs"
                  initials="ON"
                />
              </Badge.Position>
              Ola Nordmann
            </Dropdown.Button>
          </Dropdown.Item>
          <Dropdown.Item>
            <Dropdown.Button>
              <Avatar
                data-size="xs"
                data-color="support1"
                aria-label="Sogndal Kommune"
              >
                <BriefcaseIcon />
              </Avatar>
              Sogndal kommune
            </Dropdown.Button>
          </Dropdown.Item>
        </Dropdown.List>
      </Dropdown>
    </Dropdown.TriggerContext>
  ),
};

export const AsLink: Story = {
  args: {
    'aria-label': 'Ola Nordmann',
  },
  render: (args) => (
    <Link
      href="https://www.udir.no/"
      style={{ display: 'flex', gap: 'var(--ds-size-2)', alignItems: 'center' }}
    >
      <Avatar {...args} />
      Ola Nordmann
    </Link>
  ),
};
