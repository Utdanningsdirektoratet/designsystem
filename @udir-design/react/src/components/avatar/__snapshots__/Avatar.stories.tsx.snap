// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`As Link 1`] = `
<a
  href="https://www.udir.no/"
  style="display: flex; gap: var(--ds-size-2); align-items: center;"
>
  <span
    data-variant="circle"
    role="img"
    aria-label="Ola Nordmann"
  >
  </span>
  Ola Nordmann
</a>
`;

exports[`Color Variants 1`] = `
<span
  data-variant="circle"
  role="img"
  aria-label="color neutral"
  data-color="neutral"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="color accent"
  data-color="accent"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="color support1"
  data-color="support1"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="color support2"
  data-color="support2"
>
</span>
`;

exports[`In Dropdown 1`] = `
<button
  data-variant="tertiary"
  type="button"
  popovertarget="«r0»"
>
  <span
    data-variant="circle"
    role="img"
    aria-label="Ola Nordmann"
    data-size="sm"
  >
    ON
  </span>
  Velg Profil
</button>
<div
  id="«r0»"
  popover="manual"
  data-variant="default"
  data-size="md"
  data-placement="bottom"
  style="--ds-popover-arrow-x: <removed>; --ds-popover-arrow-y: <removed>; translate: <removed>;"
>
  <ul>
    <li>
      <button
        data-variant="tertiary"
        type="button"
      >
        <span
          data-overlap="circle"
          data-placement="top-right"
        >
          <span
            data-variant="base"
            data-color="danger"
            data-size="sm"
          >
          </span>
          <span
            data-variant="circle"
            role="img"
            aria-label="Ola Nordmann"
            data-size="xs"
          >
            ON
          </span>
        </span>
        Ola Nordmann
      </button>
    </li>
    <li>
      <button
        data-variant="tertiary"
        type="button"
      >
        <span
          data-variant="circle"
          role="img"
          aria-label="Sogndal Kommune"
          data-size="xs"
          data-color="support1"
        >
          <svg
            xmlns="http://www.w3.org/2000/svg"
            width="1em"
            height="1em"
            fill="none"
            viewbox="0 0 24 24"
            focusable="false"
            role="img"
            aria-hidden="true"
          >
            <path
              fill="currentColor"
              fill-rule="evenodd"
              d="M2.25 7.5c0-.69.56-1.25 1.25-1.25h4.25V5c0-.966.784-1.75 1.75-1.75h5c.966 0 1.75.784 1.75 1.75v1.25h4.25c.69 0 1.25.56 1.25 1.25v5c0 .69-.56 1.25-1.25 1.25h-.25v4.75c0 .69-.56 1.25-1.25 1.25H5c-.69 0-1.25-.56-1.25-1.25v-4.75H3.5c-.69 0-1.25-.56-1.25-1.25zm7-2.5a.25.25 0 0 1 .25-.25h5a.25.25 0 0 1 .25.25v1.25h-5.5zM12 15.75a.75.75 0 0 0 .75-.75v-1.25h6v4.5H5.25v-4.5h6V15c0 .414.336.75.75.75m0-5.5a.75.75 0 0 1 .75.75v1.25h7.5v-4.5H3.75v4.5h7.5V11a.75.75 0 0 1 .75-.75"
              clip-rule="evenodd"
            >
            </path>
          </svg>
        </span>
        Sogndal kommune
      </button>
    </li>
  </ul>
</div>
`;

exports[`No Name 1`] = `
<span
  data-variant="circle"
  role="img"
  aria-label="Ola"
>
</span>
`;

exports[`Preview 1`] = `
<span
  data-variant="circle"
  role="img"
  aria-label="Ola Nordmann"
>
</span>
`;

exports[`Shape Variants 1`] = `
<span
  data-variant="circle"
  role="img"
  aria-label="variant circle"
>
</span>
<span
  data-variant="square"
  role="img"
  aria-label="variant square"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="Ola Nordman"
>
  ON
</span>
<span
  data-variant="square"
  role="img"
  aria-label="Ola Nordman"
>
  ON
</span>
`;

exports[`Sizes 1`] = `
<span
  data-variant="circle"
  data-initials="xs"
  role="img"
  aria-label="extra small"
  data-size="xs"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="extra small"
  data-size="xs"
>
</span>
<span
  data-variant="circle"
  data-initials="sm"
  role="img"
  aria-label="small"
  data-size="sm"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="small"
  data-size="sm"
>
</span>
<span
  data-variant="circle"
  data-initials="md"
  role="img"
  aria-label="medium"
  data-size="md"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="medium"
  data-size="md"
>
</span>
<span
  data-variant="circle"
  data-initials="lg"
  role="img"
  aria-label="large"
  data-size="lg"
>
</span>
<span
  data-variant="circle"
  role="img"
  aria-label="large"
  data-size="lg"
>
</span>
`;

exports[`With Image 1`] = `
<span
  data-variant="circle"
  role="img"
  aria-label="Ola Nordman"
>
  <img
    alt
    aria-hidden="true"
    src="https://images.unsplash.com/photo-1633332755192-727a05c4013d?q=80&amp;w=1480&amp;auto=format&amp;fit=crop&amp;ixlib=rb-4.0.3&amp;ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D"
  >
</span>
`;
